blogs/dns_tracking/dns_tracking.md

```diff
@@ -0,0 +1,136 @@
+
+![logo](./images/dns_tracking_logo.png)
+
+By: Julien Pinsonneau, Mehul Modi and Mohamed S. Mahmoud
+<style>body {text-align: justify}</style>
+
+In today's interconnected digital landscape, Domain Name System (DNS) tracking
+plays a crucial role in networking and security.
+DNS resolution is a fundamental process that translates human-readable domain
+names into IP addresses, enabling communication between devices and servers.
+However, this process also presents opportunities for monitoring and analysis,
+which can be achieved through innovative technologies like
+eBPF (extended Berkeley Packet Filter).
+In this blog post, we'll delve into the world of DNS tracking using eBPF
+tracepoint hooks, exploring how this powerful combination can be used for
+various purposes, including network monitoring and security enhancement.
+
+## Understanding DNS Resolution
+
+Before diving into the specifics of eBPF tracepoint hooks, let's briefly
+recap how DNS resolution works.
+When you enter a website's domain name (e.g., www.example.com) in your
+browser, your computer needs to find the corresponding IP address.
+This process involves multiple steps, including querying DNS servers,
+caching responses, obtaining the IP address to establish a connection
+and caching response for future re-occurrence of the same DNS query.
+
+## Utilizing Tracepoint Hooks for DNS Tracking
+
+Tracepoint hooks are predefined points in the Linux kernel where eBPF
+programs can be attached to capture and analyze specific events.
+For DNS tracking, we leveraged tracepoint hooks associated with DNS
+resolution processes, specifically the `tracepoint/net/net_dev_queue` tracepoint.
+Then we parse the DNS header to determine if it is a query or a response, attempt
+to correlate the query or response for specific DNS transaction, and then record the
+elapsed time to compute DNS latency. Further, DNS network flows are enriched
+to include fields, viz. DNS Id, DNS latency and DNS response codes to help
+build graphs with aggregated DNS statistics and to help filtering on specific
+fields in the Network Observability console.
+
+## Potential Use Cases
+
+DNS tracking with eBPF tracepoint hooks can serve various purposes:
+
+- Network Monitoring: Gain insights into DNS queries and responses,
+helping network administrators identify unusual patterns,
+potential bottlenecks, or performance issues.
+
+- Security Analysis: Detect suspicious DNS activities, such as domain
+name generation algorithms (DGA) used by malware,
+or identify unauthorized DNS resolutions that might indicate a security breach.
+
+- Troubleshooting: Debug DNS-related issues by tracing DNS resolution steps,
+tracking latency, and identifying misconfigurations.
+
+## How to enable DNS tracking
+
+By default DNS tracking is disabled because it requires
+`privileged` access, to enable this feature we need to create a flow
+collector object with the following fields enabled in eBPF config
+section
```

Comment:
Do you expect readers to know how to access the eBPF config section?  If not, it would be helpful to add some details about how to access it.

Reply:
the eBPF config section is the following  yaml snippet in the same sentence, do u think we need to clarify it more ?